### PR TITLE
Update convert_matrix.py

### PR DESCRIPTION
**Pull request by akshayamadhuri**
Closes: #4225

### CODE REVIEW

**Review by rossbar — COMMENTED** [review]
@akshayamadhuri I think what we want here is a docstring example in the `from_pandas` function(s) as opposed to adding a new function.

**Comment by dschult** [comment]
There are two existing functions to bring pandas data into networkx format. `from_pandas_edgelist` forms a graph from a dataframe where the dataframe is storing each edge as a "row" in the data frame. `from_pandas_adjacency` forms a graph from a dataframe where the dataframe is storing the adjacency matrix in a dataframe form. Both of these function deal with creating a graph based on edges stored in a dataframe.

This new feature should allow addition of node attributes that are stored in a pandas dataframe. Since the one-line code to do so is so short, we thought it would be best to add a note about this feature to the doc_strings of the two `from_pandas_*` functions. So, instead of making a new function, we are looking for an addition to the doc_strings. But if someone thinks it should be a new function this is a good place to discuss that.

In the doc_strings, this isn't really an example and it is not a straightforward "See Also" (which usually points to a function). This is more like a "tip for users". So I think it could go between the "See Also" section and the "Examples" section. It should say something like: "If you have node attributes stored in a dataframe ``df`` (separate from the edge info dataframe), those attributes can be loaded to a graph ``G`` using code similar to: `G.add_nodes_from((n, dict(d)) for n, d in df.iterrows())`."

Whatever the format and language is for that note, it can appear in both of the two functions described above.  Is this more clear?

**Comment by akshayamadhuri** [comment]
@dschult  are you expecting something like this with attributes and nodes

`def from_pandas_edgelist(df, source, target, **kwargs):
    G = nx.from_pandas_edgelist(df, source=source, target=target, **kwargs)
    return G`

**Comment by akshayamadhuri** [comment]
@dschult  I am trying to modify it can you help me with this
` import networkx as nx
import pandas as pd
import matplotlib.pyplot as plt
edge_data = {
    'source': [1, 2, 3],
    'target': [2, 3, 1]
}
edge_df = pd.DataFrame(edge_data)
G = nx.Graph()
G = nx.from_pandas_edgelist(edge_df, source='source', target='target')
`

**Comment by dschult** [comment]
The second code block you sent works on my machine. I get G to be an undirected triangle on nodes 1, 2, 3.
What is your question?

**Comment by akshayamadhuri** [comment]
according to this <https://github.com/networkx/networkx/pull/7018#issuecomment-1762996854>

**Review by dschult — COMMENTED** [review]
Thanks for this revampe to doc_string suggested code.
Just a couple of asks still... :}

Could you wrap the description line to about the same as what the others are (<80 if possible and <88 for sure)?

And maybe it would be even better to have the specific format of
the dataframe spelled out rather than reading from a .csv file?
One way would be to construct the DataFrame amnually.
Something like: `df_nodes = pd.DataFrame(<something simple>)`

**Comment by akshayamadhuri** [comment]
@dschult  can you review and suggest possible changes

**Comment by akshayamadhuri** [comment]
@dschult   what about using this `df_nodes = pd.DataFrame({"node_id": [1, 2, 3], "attribute1": ["A", "B", "C"]})` or I think we can change the line to something `If you have node attributes stored in a separate dataframe or file, you can load those attributes to the graph G using the following code`

**Comment by dschult** [comment]
Yes, that line to build the `df_nodes` looks good to me. :}

File: networkx/convert_matrix.py [hunk 1]
```diff
@@ -349,6 +356,13 @@ def from_pandas_edgelist(
         this column are used for the edge keys when adding edges if create_using
         is a multigraph.
 
+   If you have node attributes stored in a separate dataframe `df_nodes`, you can load those attributes to the graph `G` using the following code:
+    
+    ```
+    df_nodes = pd.DataFrame({"node_id": [1, 2, 3], "attribute1": ["A", "B", "C"]})
+    G.add_nodes_from((n, dict(d)) for n, d in df_nodes.iterrows())
+    ``` 
+
```
Comment (dschult) [inline networkx/convert_matrix.py:365]:
````suggestion
    If you have node attributes stored in a separate dataframe `df_nodes`,
    you can load those attributes to the graph `G` using the following code:

    ```
    df_nodes = pd.DataFrame({"node_id": [1, 2, 3], "attribute1": ["A", "B", "C"]})
    G.add_nodes_from((n, dict(d)) for n, d in df_nodes.iterrows())
    ``` 

````

File: networkx/convert_matrix.py [hunk 2]
```diff
@@ -171,6 +171,13 @@ def from_pandas_adjacency(df, create_using=None):
     If `df` has a single data type for each entry it will be converted to an
     appropriate Python data type.
 
+    If you have node attributes stored in a separate dataframe `df_nodes`, you can load those attributes to the graph `G` using the following code:
+
```
Comment (dschult) [inline networkx/convert_matrix.py:175]:
```suggestion
    If you have node attributes stored in a separate dataframe `df_nodes`,
    you can load those attributes to the graph `G` using the following code:

```

**Review by dschult — COMMENTED** [review]
Looks like one version only has 3 spaces in front. I've wrapped the first line while fixing that.
There's also spaces on the blank lines for some reason, so removed them too

File: networkx/convert_matrix.py [hunk 3]
```diff
@@ -349,6 +357,14 @@ def from_pandas_edgelist(
         this column are used for the edge keys when adding edges if create_using
         is a multigraph.
 
+    If you have node attributes stored in a separate dataframe `df_nodes`,
+    you can load those attributes to the graph `G` using the following code:
+
+    ```
+    df_nodes = pd.DataFrame({"node_id": [1, 2, 3], "attribute1": ["A", "B", "C"]})
+    G.add_nodes_from((n, dict(d)) for n, d in df_nodes.iterrows())
+    ```
```
Comment (dschult) [inline networkx/convert_matrix.py:366]:
```suggestion
    ```
```

**Review by dschult — COMMENTED** [review]
Sorry for the noise. fixing white space correctly hopefully this time.

**Review by dschult — APPROVED** [review]
This looks good to me!

**Review by jarrodmillman — APPROVED** [review]

**Pull request merged**